File: app/src/main/java/dev/olog/basil/presentation/recipe/RecipeFragment.java
package dev.olog.basil.presentation.recipe;

import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import javax.inject.Inject;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.lifecycle.ViewModelProvider;
import androidx.lifecycle.ViewModelProviders;
import dev.olog.basil.R;
import dev.olog.basil.app.GlideApp;
import dev.olog.basil.presentation.base.BaseFragment;
import dev.olog.basil.presentation.main.MainFragmentViewModel;
import dev.olog.basil.presentation.navigator.Navigator;

public class RecipeFragment extends BaseFragment {

    private static final String TAG = RecipeFragment.class.getSimpleName();
    private static final String ARGUMENT_ID = TAG + ".argument.id";

    public static RecipeFragment newInstance(long id){
        Bundle bundle = new Bundle();
        bundle.putLong(ARGUMENT_ID, id);
        RecipeFragment fragment = new RecipeFragment();
        fragment.setArguments(bundle);
        return fragment;
    }

    private ImageView image;

    @Inject ViewModelProvider.Factory viewModelFactory;
    @Inject Navigator navigator;

    private MainFragmentViewModel viewModel;

    private long recipeId;

    @Override
    protected void onViewBound(@NonNull View view, @Nullable Bundle savedInstanceState) {
        TextView title = view.findViewById(R.id.title);
        image = view.findViewById(R.id.image);

        recipeId = getArguments().getLong(ARGUMENT_ID);

        viewModel = ViewModelProviders.of(requireActivity(), viewModelFactory).get(MainFragmentViewModel.class);

//        viewModel.observeRecipeById(recipeId)
//                .observe(getViewLifecycleOwner(), recipe -> {
//                    title.setText(recipe.getName());
//                });

        GlideApp.with(view)
                .load(R.drawable.spinach)
                .into(image);
    }

    @Override
    public void onResume() {
        super.onResume();
        image.setOnClickListener(v -> navigator.toDetailFragment(recipeId));
    }

    @Override
    public void onPause() {
        super.onPause();
        image.setOnClickListener(null);
    }

    @Override
    protected int provideLayoutId() {
        return R.layout.fragment_recipe;
    }
}

File: presentation/src/main/java/dev/olog/basil/presentation/recipe/RecipeFragment.kt
package dev.olog.basil.presentation.recipe

import android.os.Bundle
import android.view.View
import androidx.core.os.bundleOf
import androidx.fragment.app.Fragment
import com.bumptech.glide.Glide
import dev.olog.basil.core.Recipe
import dev.olog.basil.presentation.R
import dev.olog.basil.presentation.widget.sliding.sheet.SlidingSheet
import dev.olog.basil.shared.lazyFast
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.fragment_recipe.*
import kotlin.math.abs

class RecipeFragment : Fragment(R.layout.fragment_recipe) {

    companion object {

        private const val RECIPE = "recipe"

        fun newInstance(recipe: Recipe): RecipeFragment {
            return RecipeFragment().apply {
                arguments = bundleOf(RECIPE to recipe)
            }
        }

    }

    private val recipe by lazyFast {
        requireArguments().getParcelable<Recipe>(RECIPE)!!
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        Glide.with(requireContext())
            .load(recipe.image)
            .into(image)
    }

    override fun onResume() {
        super.onResume()
        requireActivity().slidingSheet.addListener(listener)
    }

    override fun onPause() {
        super.onPause()
        requireActivity().slidingSheet.removeListener(listener)
    }

    private val listener = object : SlidingSheet.Callback {

        private val scrimAcceleration = 2f

        override fun onOffsetChanged(view: SlidingSheet, offset: Float) {
            if (offset > 0) {
                image.scrimPercentage = 0f
                return
            }
            image.scrimPercentage = abs(offset * scrimAcceleration)
        }
    }

}